src/mlpack/methods/ann/layer/nearest_interpolation_impl.hpp

```diff
@@ -149,13 +149,14 @@ void NearestInterpolationType<MatType>::Backward(
 template<typename MatType>
 void NearestInterpolationType<MatType>::ComputeOutputDimensions()
 {
-  if (this->inputDimensions.size() - 1 != scaleFactors.size()) {
-    throw std::runtime_error("Scale factors must match number of rows and columns.");
+  if (this->inputDimensions.size() < scaleFactors.size()) {
+    throw std::runtime_error("Insufficient number of input dimensions.");
```

Comment:
If this exception gets thrown, in general the user won't get a backtrace.  So it can be really helpful to provide a bit more information about where the error is in the exception message.  It can also be helpful to provide more details on what is wrong, although that is more tedious...

```suggestion
    std::ostringstream oss;
    oss << "NearestInterpolation::ComputeOutputDimensions(): input dimensions "
        << "must be at least 2 (received input with " << inputDimensions.size()
        << " dimensions)!";
    throw std::runtime_error(oss.str());
```